src/Canvas.ts

```diff
@@ -3,15 +3,27 @@ import AlphaColor from './Color/AlphaColor';
 class Canvas {
   ctx: CanvasRenderingContext2D;
   canvasElement: HTMLCanvasElement;
+  lastImageData: ImageData | null = null;
+  private _backgroundCanvas: Canvas | null = null;
 
-  constructor(canvasElement: HTMLCanvasElement) {
-    this.canvasElement = canvasElement;
+  get backgroundCanvas(): Canvas {
+    if (this._backgroundCanvas === null) {
+      throw new Error('Cannot access backgroundCanvas of a background canvas');
+    }
+    return this._backgroundCanvas;
+  }
 
+  constructor(canvasElement: HTMLCanvasElement, createBackground = true) {
```

Comment:
Let's create a CanvasManager because there's nothing i like more than... MANAGERS

Reply:
And because fuck that ! 
https://stackoverflow.com/questions/1866794/naming-classes-how-to-avoid-calling-everything-a-whatevermanager
Managers will conquiers the world